### PR TITLE
Fix snakecase header consistency

### PR DESCRIPTION
**Pull request by kbirk**
I originally touch a lot more files than I intended, and my editor injected `final` in them, even after reverting. 

Tagged in some quick migrations :)

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/SnakeCaseRequestControllerAdvice.java [hunk 1]
```diff
@@ -19,19 +20,18 @@
 import software.uncharted.terarium.hmiserver.annotations.AMRPropertyNamingStrategy;
 
 @RestControllerAdvice
+@RequiredArgsConstructor
 @Slf4j
 public class SnakeCaseRequestControllerAdvice extends RequestBodyAdviceAdapter {
-	private ObjectMapper snakecaseMapper;
-	private ObjectMapper camelcaseMapper;
+	private final ObjectMapper snakecaseMapper;
```
Comment (kbirk) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/SnakeCaseRequestControllerAdvice.java:26]: Use springboot injected ObjectMapper here

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/SnakeCaseResponseControllerAdvice.java [hunk 2]
```diff
@@ -16,40 +17,40 @@
...
 public class SnakeCaseResponseControllerAdvice implements ResponseBodyAdvice {
-	private ObjectMapper mapper;
+	private final ObjectMapper mapper;
 
 	@PostConstruct
 	public void init() {
-		mapper = new ObjectMapper()
-				.setPropertyNamingStrategy(
-						new AMRPropertyNamingStrategy(new PropertyNamingStrategies.SnakeCaseStrategy()));
+		mapper.setPropertyNamingStrategy(
```
Comment (kbirk) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/SnakeCaseResponseControllerAdvice.java:26]: Use springboot injected ObjectMapper

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java [hunk 3]
```diff
@@ -38,7 +38,8 @@ public class Simulation extends TerariumAsset {
 	private static final long serialVersionUID = 5467224100686908152L;
 
 	@JsonAlias("execution_payload")
-	@Column(columnDefinition = "text")
+	@Type(JsonType.class)
+	@Column(columnDefinition = "json")
```
Comment (kbirk) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/Simulation.java:42]: Update to json, missed that earlier. There is a migration to migrate it.

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/SimulationUpdate.java [hunk 4]
```diff
@@ -2,54 +2,39 @@
...
-	@PrePersist
-	protected void onCreate() {
-		this.createdOn = this.createdOn != null
-				? this.createdOn
-				: Timestamp.from(ZonedDateTime.now(ZoneId.systemDefault()).toInstant());
-	}
-
-	@Convert(converter = JsonConverter.class)
-	@Column(columnDefinition = "text")
+	@Type(JsonType.class)
```
Comment (kbirk) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/simulation/SimulationUpdate.java:29]: Update to json, missed that earlier. There is a migration to migrate it.

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/notification/NotificationEvent.java [hunk 5]
```diff
@@ -40,8 +40,8 @@ public class NotificationEvent extends TerariumEntity {
 	@Column(columnDefinition = "TIMESTAMP WITH TIME ZONE")
 	private Timestamp acknowledgedOn = null;
 
-	@Convert(converter = JsonConverter.class)
-	@Column(columnDefinition = "text")
+	@Type(JsonType.class)
```
Comment (kbirk) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/notification/NotificationEvent.java:43]: Update to json, missed that earlier. There is a migration to migrate it.

**Review by dvince2 — APPROVED** [review]

**Pull request merged**